From 7a3ecba391a8b0f21025ce18b4742932f4f2a2ca Mon Sep 17 00:00:00 2001
From: Rob Paisley <paisley@rpaisley.com>
Date: Sun, 31 Jan 2010 11:24:47 -0800
Subject: cother/iother Segfault patch

Applied-Upstream: 123abf526df35cb6eed59e13516bea653a065873

There is a bug in :cother and :iother.
If are no windows and you issue either :cother, or
:iother ratpoison segfaults.  This patch resolves the issue.
---
 src/actions.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/actions.c b/src/actions.c
index b6fb092..47d0227 100644
--- a/src/actions.c
+++ b/src/actions.c
@@ -5725,10 +5725,11 @@ cmd_iprev (int interactive, struct cmdarg **args)
 cmdret *
 cmd_cother (int interactive UNUSED, struct cmdarg **args UNUSED)
 {
-  rp_window *cur, *w;
+  rp_window *cur, *w = NULL;
 
   cur = current_window();
-  w = group_last_window_by_class (rp_current_group, cur->res_class);
+  if (cur)
+    w = group_last_window_by_class (rp_current_group, cur->res_class);
 
   if (!w)
     return cmdret_new (RET_FAILURE, "%s", MESSAGE_NO_OTHER_WINDOW);
@@ -5741,10 +5742,11 @@ cmd_cother (int interactive UNUSED, struct cmdarg **args UNUSED)
 cmdret *
 cmd_iother (int interactive UNUSED, struct cmdarg **args UNUSED)
 {
-  rp_window *cur, *w;
+  rp_window *cur, *w = NULL;
 
   cur = current_window();
-  w = group_last_window_by_class_complement (rp_current_group, cur->res_class);
+  if (cur)
+    w = group_last_window_by_class_complement (rp_current_group, cur->res_class);
 
   if (!w)
     return cmdret_new (RET_FAILURE, "%s", MESSAGE_NO_OTHER_WINDOW);
